core/server-entry.ts

```diff
@@ -56,28 +58,26 @@ function getHostFromHeader (headers: string[]): string {
 }
 
 export default async context => {
-  const { app, router, store, initialState } = await createApp(context, context.vs && context.vs.config ? context.vs.config : buildTimeConfig)
+  let storeCode = context.vs.storeCode
+  if (config.storeViews.multistore === true) {
+    if (!storeCode) { // this is from url
+      const currentRoute = Object.assign({ path: queryString.parseUrl(context.url).url/* this gets just the url path part */, host: context.server.request.headers.host })
+      storeCode = storeCodeFromRoute(currentRoute)
+    }
+  }
+  const { app, router, store, initialState } = await createApp(context, context.vs && context.vs.config ? context.vs.config : buildTimeConfig, storeCode)
+
+  RouterManager.flushRouteQueue(router)
   context.initialState = initialState
   return new Promise((resolve, reject) => {
     context.output.cacheTags = new Set<string>()
     const meta = (app as any).$meta()
     router.push(context.url)
     context.meta = meta
     router.onReady(() => {
-      if (config.storeViews.multistore === true) {
-        let storeCode = context.vs.storeCode // this is from http header or env variable
-        if (storeCode === undefined && router.currentRoute) { // this is from url
-          const host = getHostFromHeader(context.server.request.headers)
-          const currentRoute = Object.assign({}, router.currentRoute, { host })
-          storeCode = storeCodeFromRoute(currentRoute)
-        }
-        if (storeCode !== '' && storeCode !== null) {
-          prepareStoreView(storeCode)
-        }
-      }
       const matchedComponents = router.getMatchedComponents()
-      if (!matchedComponents.length) {
-        return reject(new HttpError('No components matched', 404))
+      if (!matchedComponents.length || !matchedComponents[0]) {
```

Comment:
I don't get the ` !matchedComponents[0]` part of this condition?

Reply:
During development there were cases when getMatchedComponents returned `[undefined]` - an array with a single undefined item, instead of an empty array